app/controllers/galleries_controller.rb

```diff
@@ -101,8 +101,6 @@ def update
   end
 
   def icon
-    find_gallery if params[:id] != '0'
```

Comment:
`setup_new_icons` is called before `#icon`, which includes this call, so this is redundant.